src/App.tsx

```diff
@@ -1,61 +1,123 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { useState } from 'react';
+import { Todo } from './types/Todo';
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosFromServer);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const getTodoWithUsers = () => {
+    return todos.map(todo => {
+      const user = usersFromServer.find(
+        currUser => currUser.id === todo.userId,
+      );
+
+      if (!user) {
+        throw new Error('!user');
+      }
+
+      return {
+        ...todo,
+        user,
+      };
+    });
+  };
+
+  const onTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const onUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const onSubmit = (newTodo: Todo) => {
+    const newId = Math.max(...todos.map(todo => todo.id)) + 1;
+    const currentTodo = {
+      ...newTodo,
+      id: newId,
+    };
+
+    setTodos(currentTodos => [...currentTodos, currentTodo]);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onSubmit({
+      id: 0,
+      title: title,
+      completed: false,
+      userId: userId,
+      user: usersFromServer.find(user => user.id === userId) || null,
+    });
+
+    setTitle('');
+    setUserId(0);
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
       <form action="/api/todos" method="POST">
```

Comment:
The form should use the `onSubmit` event handler instead of the `onClick` event on the submit button. This ensures that the form submission is handled correctly and prevents default form submission behavior.